sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaRecord.java

```diff
@@ -28,7 +27,7 @@
  * KafkaRecord contains key and value of the record as well as metadata for the record (topic name,
  * partition id, and offset).
  */
-public class KafkaRecord<K, V> implements Serializable {
+public class KafkaRecord<K, V> {
```

Comment:
`KafkaRecord` should be Serializable. Or something deeply changed in the IO, Isn't this the payload of the Reader (aka what is sent in the network?)

Reply:
For payload we use coders, don't we? `KafkaRecordCoder` is used for `KafkaRecord` in this case.

Reply:
Yes you are right this is `Coder`'s job, good point.